base/broadcast.jl

```diff
@@ -961,6 +966,11 @@ broadcast_unalias(::Nothing, src) = src
 # * "extrudes" the arguments where it is advantageous to pre-compute the broadcasted indices
 @inline preprocess(dest, bc::Broadcasted{Style}) where {Style} = Broadcasted{Style}(bc.f, preprocess_args(dest, bc.args), bc.axes)
 preprocess(dest, x) = extrude(broadcast_unalias(dest, x))
+function preprocess(dest::AbstractArray, x::AbstractArray)
```

Comment:
I'm not entirely sure this is where we should do this... could you elaborate on why this was needed?

Reply:
I moved this into indexing and pushed it.  Haven't run the whole test suite locally, but I think this should be more robust.  We'll see...

Reply:
Thanks, Matt! That looks sensible to me.